src/main/java/io/supertokens/webserver/InputParser.java

```diff
@@ -137,7 +137,11 @@ public static String parseStringOrThrowError(JsonObject element, String fieldNam
             if (!stringified.contains("\"")) {
                 throw new Exception();
             }
-            return ((JsonObject) element).get(fieldName).getAsString();
+            String s = element.get(fieldName).getAsString().trim();
+            if (s.contains("@")) {
+                s = s.toLowerCase();
+            }
```

Comment:
We can't assume that if there is an `@`, it must be for an email input. Instead, we should use the full email regex to check, and if it passes that, only then we should do the toLowerCase.

Reply:
Done